qiita_db/test/test_sql.py

```diff
@@ -162,39 +176,49 @@ def test_artifact_ancestry_leaf_multiple_parents(self):
         parent2 = self._create_child_artifact([root])
         child = self._create_child_artifact([parent1, parent2])
 
-        sql = "SELECT * FROM qiita.artifact_ancestry(%s)"
-        obs = self.conn_handler.execute_fetchall(sql, [child.id])
+        with qdb.sql_connection.TRN:
+            sql = "SELECT * FROM qiita.artifact_ancestry(%s)"
+            qdb.sql_connection.TRN.add(sql, [child.id])
+            obs = qdb.sql_connection.TRN.execute_fetchindex()
         exp = [[child.id, parent1.id], [child.id, parent2.id],
                [parent1.id, root.id], [parent2.id, root.id]]
         self.assertCountEqual(obs, exp)
 
     def test_artifact_ancestry_middle(self):
         """Correctly returns the ancestry of an artifact in the middle of the
         DAG"""
-        sql = "SELECT * FROM qiita.artifact_ancestry(%s)"
-        obs = self.conn_handler.execute_fetchall(sql, [2])
+        with qdb.sql_connection.TRN:
+            sql = "SELECT * FROM qiita.artifact_ancestry(%s)"
+            qdb.sql_connection.TRN.add(sql, [2])
+            obs = qdb.sql_connection.TRN.execute_fetchindex()
```

Comment:
It looks like some of these execute_fetchindex() calls are replacing execute_fetchall() calls from conn_handler. I doubt most of these query results are large, but would it be better to replace them with execute_fetchall() calls instead?

Reply:
This comment is moot, I had it wrong: the fetch index() isn't wrapping a paging mechanism, it's doing fetchall() underneath in _execute(). The optional idx parameter is to reference the results of a particular query in a transaction block.